### PR TITLE
Add Red Hat ecosystem

### PR DESCRIPTION
**Pull request by jasinner**
Add the `Red Hat` ecosystem, see https://github.com/google/osv.dev/issues/1404

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -667,6 +678,7 @@ The defined ecosystems are:
 | `Photon OS` | The Photon OS package ecosystem; the `name` is the name of the RPM package. The ecosystem string must have a `:<RELEASE-NUMBER>` suffix to scope the package to a particular Photon OS release. Eg `Photon OS:3.0`. |
 | `Pub` | The package manager for the Dart ecosystem; the `name` field is a Dart package name. |
 | `PyPI` | the Python PyPI ecosystem; the `name` field is a [normalized](https://www.python.org/dev/peps/pep-0503/#normalized-names) PyPI package name.  |
++| `Red Hat` | The Red Hat package ecosystem; the `name` field is the name of a binary RPM. The ecosystem string is a translation of Red Hat [Common Platform Enumerations](https://cpe.mitre.org/) (CPEs). The suffix of Red Hat CPEs are used as ecosystem idenfiers with the `cpe/:[oa]:(redhat)` prefixes replaced with `Red Hat`. Red Hat ecosystem identifiers can be used to idenfity vulnerability RPMs installed on a Red Hat system as explained [here](https://www.redhat.com/en/blog/how-accurately-match-oval-security-data-installed-rpms).  |
```
Comment (andrewpollock) [inline docs/schema.md:681]: Fix spelling and grammar
```suggestion
+| `Red Hat` | The Red Hat package ecosystem; the `name` field is the name of a binary RPM. The ecosystem string is a translation of Red Hat [Common Platform Enumerations](https://cpe.mitre.org/) (CPEs). The suffix of Red Hat CPEs are used as ecosystem identifiers with the `cpe/:[oa]:(redhat)` prefixes replaced with `Red Hat`. Red Hat ecosystem identifiers can be used to identify vulnerable RPMs installed on a Red Hat system as explained [here](https://www.redhat.com/en/blog/how-accurately-match-oval-security-data-installed-rpms).  |
```

Comment (andrewpollock) [reply docs/schema.md:681]: Is the "binary RPM" part accurate? I thought I saw source RPMs getting mentioned in the conversion as well?

**Review by andrewpollock — CHANGES_REQUESTED** [review]
Please also modify the prefix and ecosystem validation at https://github.com/ossf/osv-schema/blob/4f79ec8773c0a89ffd88643ccc88772319d62be6/validation/schema.json#L299-L310

File: README.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ This is the repository for the Open Source Vulnerability schema (OSV Schema), wh
 - [PyPI Advisory Database](https://github.com/pypa/advisory-database)
 - [Python Software Foundation Database](https://github.com/psf/advisory-database)
 - [RConsortium Advisory Database](https://github.com/RConsortium/r-advisory-database)
+- [Red Hat](https://access.redhat.com/security/data)
```
Comment (andrewpollock) [inline README.md:20]: Looks like this should be what it is (at least it works and is consistent with docs/schema.md)
```suggestion
- [Red Hat](https://security.access.redhat.com/data)
```

**Review by andrewpollock — APPROVED** [review]
I think this LGTM, with one inconsistency to address.

File: README.md (hunk 2 above)
Comment (jasinner) [reply README.md]: Fixed, thanks.

**Comment by jasinner** [comment]
> Please also modify the prefix and ecosystem validation at
> 
> https://github.com/ossf/osv-schema/blob/4f79ec8773c0a89ffd88643ccc88772319d62be6/validation/schema.json#L299-L310

I added that. Thanks.

File: validation/schema.json
Comment (andrewpollock) [inline validation/schema.json:1]: Please also add the prefix validation as well

File: docs/schema.md [hunk 3]
```diff
@@ -667,6 +678,7 @@ The defined ecosystems are:
 | `Photon OS` | The Photon OS package ecosystem; the `name` is the name of the RPM package. The ecosystem string must have a `:<RELEASE-NUMBER>` suffix to scope the package to a particular Photon OS release. Eg `Photon OS:3.0`. |
 | `Pub` | The package manager for the Dart ecosystem; the `name` field is a Dart package name. |
 | `PyPI` | the Python PyPI ecosystem; the `name` field is a [normalized](https://www.python.org/dev/peps/pep-0503/#normalized-names) PyPI package name.  |
++| `Red Hat` | The Red Hat package ecosystem; the `name` field is the name of a binary RPM. The ecosystem string is a translation of Red Hat [Common Platform Enumerations](https://cpe.mitre.org/) (CPEs). The suffix of Red Hat CPEs are used as ecosystem identifiers with the `cpe/:[oa]:(redhat)` prefixes replaced with `Red Hat`. Red Hat ecosystem identifiers can be used to identify vulnerable RPMs installed on a Red Hat system as explained [here](https://www.redhat.com/en/blog/how-accurately-match-oval-security-data-installed-rpms).  |
```
Comment (oliverchang) [inline docs/schema.md:681]:
```suggestion
+| `Red Hat` | The Red Hat package ecosystem; the `name` field is the name of a binary RPM. The ecosystem string has a `:<CPE>` suffix to scope the RPM to a specific Red Hat product stream. `<CPE>` is a translation of a Red Hat [Common Platform Enumerations](https://cpe.mitre.org/) (CPE) with the `cpe/:[oa]:(redhat):` prefix removed (for example, `Red Hat:rhel_aus:8.4::appstream`). Red Hat ecosystem identifiers can be used to identify vulnerable RPMs installed on a Red Hat system as explained [here](https://www.redhat.com/en/blog/how-accurately-match-oval-security-data-installed-rpms).  |
```

**Review by oliverchang — CHANGES_REQUESTED** [review]
nice! Just some minor rewording suggestions to make this more consistent with other ecosystem definitions and to make things slightly clearer.

File: docs/schema.md (hunk 3 above)
Comment (jasinner) [reply docs/schema.md]: Thanks, I tweaked your example slightly by adding also the equivalent `rhel_aus:8.4::appstream` CPE value.

I also added 'source' RPM in the explanation of the `name` field.

**Review by oliverchang — APPROVED** [review]

File: validation/schema.json
Comment (andrewpollock) [reply validation/schema.json:1]: FTFY

**Pull request merged**